test/resource_unittest.cc

```diff
@@ -0,0 +1,68 @@
+#include "../src/resource.h"
+#include "../src/json.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(ResourceTest, Type) {
+  std::map<std::string, std::string> m;
+  google::MonitoredResource mr("type", m);
```

Comment:
[Optional] Let's go with something other than "type", to make sure the field name does not accidentally leak into the value. I would propose something like "some_resource".

Reply:
Good idea, changed.